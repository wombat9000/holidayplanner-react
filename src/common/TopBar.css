.TopBar {
    text-align: left;
    display: flex;
    flex-wrap: nowrap;
}

.flex {
    flex-grow: 1;
}