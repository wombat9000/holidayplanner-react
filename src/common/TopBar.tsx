import {AppBar, Toolbar, Typography} from '@material-ui/core';
import * as React from 'react';


class TopBar extends React.Component<Props> {
    constructor(props: any) {
        super(props);
    }

    public render() {
        return (
                <AppBar position="static">
                    <Toolbar variant="dense">
                        <Typography variant="title" color="inherit">
                            {this.props.title}
                        </Typography>
                    </Toolbar>
                </AppBar>
        );
    }
}

interface Props {
    title: string;
}

export default TopBar;
