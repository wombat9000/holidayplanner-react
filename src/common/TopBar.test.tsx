import {shallow, ShallowWrapper} from 'enzyme';
import TopBar from './TopBar';
import * as React from 'react';

describe('TopBar', () => {

    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<TopBar title="someTitle"/>)
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });
});