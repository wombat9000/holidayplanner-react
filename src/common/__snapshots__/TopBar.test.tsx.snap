// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopBar
    title="someTitle"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Toolbar)
        variant="dense"
      >
        <WithStyles(Typography)
          color="inherit"
          variant="title"
        >
          someTitle
        </WithStyles(Typography)>
      </WithStyles(Toolbar)>,
      "position": "static",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Typography)
          color="inherit"
          variant="title"
        >
          someTitle
        </WithStyles(Typography)>,
        "variant": "dense",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "someTitle",
          "color": "inherit",
          "variant": "title",
        },
        "ref": null,
        "rendered": "someTitle",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Toolbar)
          variant="dense"
        >
          <WithStyles(Typography)
            color="inherit"
            variant="title"
          >
            someTitle
          </WithStyles(Typography)>
        </WithStyles(Toolbar)>,
        "position": "static",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Typography)
            color="inherit"
            variant="title"
          >
            someTitle
          </WithStyles(Typography)>,
          "variant": "dense",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "someTitle",
            "color": "inherit",
            "variant": "title",
          },
          "ref": null,
          "rendered": "someTitle",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
